src/main/java/org/sonatype/repository/vgo/internal/util/VgoPathUtils.java

```diff
@@ -88,4 +89,18 @@ private String match(final TokenMatcher.State state, final String name) {
   public TokenMatcher.State matcherState(final Context context) {
     return context.getAttributes().require(TokenMatcher.State.class);
   }
+
+  /**
+   * Returns the module name and version from a given {@link TokenMatcher.State}
+   *
+   * @param state
+   * @return {@link VgoAttributes}
+   */
+  public VgoAttributes getAttributesFromMatcherState(final TokenMatcher.State state) {
```

Comment:
I know some of the facet stuff is not fun to unit test but methods like this can absolutely be tested.

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-vgo/pull/5/commits/47ee2def4ef1918fbec752f12ecf59d259cad14d